src/index.cpp

```diff
@@ -7486,29 +7492,133 @@ void Index::batch_embed_fields(std::vector<index_record*>& records,
                 LOG(ERROR) << "Error: " << error_msg;
                 return;
             }
-            embeddings = embedder_op.get()->batch_embed(values);
-        } else {
+            embeddings_image = embedder_op.get()->batch_embed(values_image);
+        }
+
+        if(!values_text.empty()) {
             auto embedder_op = embedder_manager.get_text_embedder(field.embed[fields::model_config]);
             if(!embedder_op.ok()) {
                 LOG(ERROR) << "Error while getting embedder for model: " << field.embed[fields::model_config];
                 LOG(ERROR) << "Error: " << embedder_op.error();
                 return;
             }
-            embeddings = embedder_op.get()->batch_embed(values, remote_embedding_batch_size, remote_embedding_timeout_ms,
+            embeddings_text = embedder_op.get()->batch_embed(values_text, remote_embedding_batch_size, remote_embedding_timeout_ms,
                                                             remote_embedding_num_tries);
         }
 
-        for(size_t i = 0; i < embeddings.size(); i++) {
-            auto& embedding_res = embeddings[i];
-            if(!embedding_res.success) {
-                values_to_embed[i].first->embedding_res = embedding_res.error;
-                values_to_embed[i].first->index_failure(embedding_res.status_code, "");
-                continue;
+        for(auto& record: records_to_index) {
+            size_t count = 0;
+            if(!values_to_embed_text.empty()) {
+                for(size_t i = 0; i < values_to_embed_text.size(); i++) {
+                    auto& value_to_embed = values_to_embed_text[i];
+                    if(record == value_to_embed.first) {
+                        if(!value_to_embed.first->embedding_res.empty()) {
+                            continue;
+                        }
+                        if(!embeddings_text[i].success) {
+                            value_to_embed.first->embedding_res = embeddings_text[i].error;
+                            value_to_embed.first->index_failure(embeddings_text[i].status_code, "");
```

Comment:
Instead of nesting code under `else {` make this `if` condition have a `continue;`